src/Model/Selectors.re

```diff
@@ -58,6 +58,20 @@ let getActiveBuffer = (state: State.t) => {
   };
 };
 
+let withActiveBufferAndFileType = (state: State.t, f) => {
+  let buffer = getActiveBuffer(state);
+
+  switch (buffer) {
+  | None => ()
+  | Some(buf) =>
+    let fileType = Buffer.getFileType(buf);
+    switch (fileType) {
+    | None => ()
+    | Some(ft) => f(buf, ft)
+    };
+  };
+};
```

Comment:
`Option.map` + `Option.iter` would shorten this significantly.

Reply:
Good idea! I added `Option.iter` 👍 

One thing that was a bit awkward is we needed to keep the `buf` around along with the filetype `ft` - I'm not sure if I did it in the cleanest way.

Reply:
I have another change that depends on this, so I'll bring it in... but if you happen to have feedback I'll incorporate in a separate PR. Thanks for the suggestion @glennsl !